stix2validator/validator.py

```diff
@@ -110,8 +111,38 @@ def as_json(self):
         return json.dumps(self.as_dict())
 
 
-class ValidationResults(BaseResults):
-    """Results of JSON schema validation.
+class FileValidationResults(BaseResults):
+    """
+    Represents validation results for a file.  This entails potentially
+    several STIX object results, since a file may contain a list of STIX
+    objects.
+    """
+    def __init__(self, is_valid=False, filepath=None, object_results=None, fatal=None):
+        """
+        Initialize this instance.
+        :param is_valid: Whether the overall result is valid
+        :param filepath: Which file was validated
+        :param object_results: Individual object validation results
+        :param fatal: A non-validation-related fatal error
+        """
+        super(FileValidationResults, self).__init__(is_valid)
+        self.filepath = filepath
+        self.object_results = object_results or []
+        self.fatal = fatal
+
+    def as_dict(self):
+        d = super(FileValidationResults, self).as_dict()
+        d.update(
+            filepath=self.filepath,
+            object_results=[object_result.as_dict() for object_result in self.object_results],
+            fatal=self.fatal.as_dict()
+        )
+
+        return d
+
+
+class ObjectValidationResults(BaseResults):
```

Comment:
I have mixed feelings about calling this `ObjectValidationResults`, particularly since it will in practice often be used to validate a Bundle. Particularly since passing in a single bundle to `validate` will result in a list of ObjectValidationResults, but this list will only have a single item for the entire bundle rather than a one result per item in the bundle's `objects`.

Reply:
I haven't changed the name, but a single bundle would now produce a single `ObjectValidationResults` instead of a list.  Should bundles themselves be validated according to the bundle schema, in addition to contained objects?  There looks to be a schema for them, but right now the contained objects are validated but not the bundle itself.